src/generic/MPoly.jl

```diff
@@ -115,6 +115,41 @@ function derivative(f::MPoly{T}, x::MPoly{T}) where {T <: RingElement}
    return(derivative)
 end
 
+
+
+doc"""
+    change_base_ring(p::AbstractAlgebra.Generic.MPoly{T}, g) where {T <: RingElement}
+> Returns the polynomial obtained by applying g to the coefficients of p.
+"""
+function change_base_ring(p::AbstractAlgebra.Generic.MPoly{T}, g) where {T <: RingElement}
+   vars_parent = vars(p.parent)
+
+   n = p.parent.num_vars
+   exps = p.exps
+
+   size_exps = size(p.exps)
+   if (p.parent.ord != :lex)
+      exps = exps[2:size_exps[1],:]
+   end
+   if (p.parent.ord == :degrevlex)
+      exps = exps[end:-1:1,:]
+   end
+
+   new_p = g(zero(p.parent.base_ring))
+   new_base_ring = parent(new_p)
+   new_polynomial_ring, gens_new_polynomial_ring = PolynomialRing(new_base_ring, [string(v) for v in vars_parent])
+
+   coeffs = p.coeffs
+   for i=1:length(p)
+      prod = g(coeffs[i])
```

Comment:
I'm not completely sure, but I think it would be more efficient to put `p.coeffs[1]` here, so that you can avoid allocating the array at line 142.